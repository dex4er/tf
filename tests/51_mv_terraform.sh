#!/usr/bin/env bash

set -euo pipefail

cd "$(dirname "$0")"
test=$(basename "$0" .sh)

export TERRAFORM_PATH=${test##*_}

rm -rf tmp/$test
mkdir -p tmp/$test
cp ../demo.tf tmp/$test
pushd tmp/$test >/dev/null

{
  set -x
  ../../../tf init
  ../../../tf destroy -auto-approve
  ../../../tf import time_sleep.this["1s"] 1s,1s
  ../../../tf list
  ../../../tf mv time_sleep.this["1s"] time_sleep.this["2s"]
  ../../../tf list
} 2>&1 | ../../sanitize.sh >>tf.out

diff -u ../../$test.out tf.out

popd >/dev/null

rm -rf tmp/$test
