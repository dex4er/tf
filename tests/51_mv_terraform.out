+ ../../../tf init
- Installing hashicorp/XXX vX.X.X...
- Installed hashicorp/XXX vX.X.X (signed by HashiCorp)
- Installing hashicorp/XXX vX.X.X...
- Installed hashicorp/XXX vX.X.X (signed by HashiCorp)
[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m
[0m+ ../../../tf destroy -auto-approve
[0m[1m[32mNo changes.[0m[1m No objects need to be destroyed.[0m
[0mEither you have not created any objects yet or the existing objects were
already deleted outside of Terraform.
[0m[1m[32m
Destroy complete! Resources: 0 destroyed.
[0m+ ../../../tf import 'time_sleep.this[1s]' 1s,1s
[90m&X/X [0m[1mtime_sleep.this["Xs"]: Importing from ID "1s,1s"...[0m
[0m[1m[32mtime_sleep.this["Xs"]: Import prepared![0m
[34m^X [0m[90m&X/X [0m[1mtime_sleep.this["Xs"]: Refreshing state...[0m
[0m[32m
Import successful!
[0m# time_sleep.this["Xs"]:
resource "time_sleep" "this" {
    create_duration  = "1s"
    destroy_duration = "1s"
    id               = "XXXX-XX-XXTXX:XX:XXZ"
    triggers         = {}
}
+ ../../../tf list
time_sleep.this["1s"]
+ ../../../tf mv 'time_sleep.this[1s]' 'time_sleep.this[2s]'
Move "time_sleep.this[\"1s\"]" to "time_sleep.this[\"2s\"]"
Successfully moved 1 object(s).
+ ../../../tf list
time_sleep.this["2s"]
